core/dbt/cli/requires.py

```diff
@@ -97,7 +98,16 @@ def wrapper(*args, **kwargs):
 
         try:
             result, success = func(*args, **kwargs)
-        except Exception as e:
+        except DbtException as e:
+            fire_event(MainEncounteredError(exc=str(e)))
+            raise HandledExit(
+                result=e.result,
+                success=False,
+                exit_code=ExitCodes.ModelError.value,
+            )
+        except BaseException as e:
+            fire_event(MainEncounteredError(exc=str(e)))
+            fire_event(MainStackTrace(stack_trace=traceback.format_exc()))
```

Comment:
Ah - does this decorator not get called during parsing (which happens within `@requires.manifest`)?

I'm noticing that we're still just bubbling up the full exception/stacktrace if I have (e.g.) invalid yaml in my project:
```
 $ dbt parse
18:47:04  Running with dbt=1.5.0-b4
Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt-core/core/dbt/clients/yaml_helper.py", line 56, in load_yaml_text
    return safe_load(contents)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/clients/yaml_helper.py", line 51, in safe_load
    return yaml.load(contents, Loader=SafeLoader)
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/yaml/__init__.py", line 81, in load
    return loader.get_single_data()
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "yaml/_yaml.pyx", line 673, in yaml._yaml.CParser.get_single_node
  File "yaml/_yaml.pyx", line 687, in yaml._yaml.CParser._compose_document
  File "yaml/_yaml.pyx", line 731, in yaml._yaml.CParser._compose_node
  File "yaml/_yaml.pyx", line 845, in yaml._yaml.CParser._compose_mapping_node
  File "yaml/_yaml.pyx", line 729, in yaml._yaml.CParser._compose_node
  File "yaml/_yaml.pyx", line 806, in yaml._yaml.CParser._compose_sequence_node
  File "yaml/_yaml.pyx", line 731, in yaml._yaml.CParser._compose_node
  File "yaml/_yaml.pyx", line 847, in yaml._yaml.CParser._compose_mapping_node
  File "yaml/_yaml.pyx", line 860, in yaml._yaml.CParser._parse_next_event
yaml.scanner.ScannerError: mapping values are not allowed in this context
  in "<unicode string>", line 3, column 15

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/schemas.py", line 106, in yaml_from_file
    return load_yaml_text(source_file.contents or "", source_file.path)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/clients/yaml_helper.py", line 63, in load_yaml_text
    raise dbt.exceptions.DbtValidationError(error)
dbt.exceptions.DbtValidationError: Runtime Error
  Syntax error near line 3
  ------------------------------
  1  | models:
  2  |   - name: customers
  3  |       bad_yaml: something_else

  Raw Error:
  ------------------------------
  mapping values are not allowed in this context
    in "<unicode string>", line 3, column 15

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt-core/env/bin/dbt", line 33, in <module>
    sys.exit(load_entry_point('dbt-core', 'console_scripts', 'dbt')())
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/env/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/cli/requires.py", line 88, in wrapper
    return func(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/cli/requires.py", line 165, in wrapper
    return func(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/cli/requires.py", line 187, in wrapper
    return func(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/cli/requires.py", line 213, in wrapper
    return func(*args, **kwargs)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/cli/requires.py", line 240, in wrapper
    manifest = ManifestLoader.get_full_manifest(
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/manifest.py", line 226, in get_full_manifest
    manifest = loader.load()
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/manifest.py", line 272, in load
    file_reader.read_files()
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/read_files.py", line 193, in read_files
    self.read_files_for_project(project, file_types)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/read_files.py", line 200, in read_files_for_project
    project_files[file_type_info["parser"]] = read_files_for_parser(
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/read_files.py", line 157, in read_files_for_parser
    source_files = get_source_files(
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/read_files.py", line 146, in get_source_files
    file = load_source_file(fp, parse_file_type, project.project_name, saved_files)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/read_files.py", line 79, in load_source_file
    dfy = yaml_from_file(source_file)
  File "/Users/jerco/dev/product/dbt-core/core/dbt/parser/schemas.py", line 108, in yaml_from_file
    raise YamlLoadError(
dbt.exceptions.YamlLoadError: Parsing Error
  Error reading test: customer.yml - Runtime Error
    Syntax error near line 3
    ------------------------------
    1  | models:
    2  |   - name: customers
    3  |       bad_yaml: something_else

    Raw Error:
    ------------------------------
    mapping values are not allowed in this context
      in "<unicode string>", line 3, column 15
```

Versus the behavior in v1.4:
```
 $ dbt parse
18:46:21  Running with dbt=1.4.5
18:46:21  Start parsing.
18:46:21  Dependencies loaded
18:46:22  Unable to do partial parsing because of a version mismatch
18:46:22  ManifestLoader created
18:46:22  Encountered an error:
Parsing Error
  Error reading test: customer.yml - Runtime Error
    Syntax error near line 3
    ------------------------------
    1  | models:
    2  |   - name: customers
    3  |       bad_yaml: something_else

    Raw Error:
    ------------------------------
    mapping values are not allowed in this context
      in "<unicode string>", line 3, column 15
```

I understand that wasn't in the initial scope of #7118, so we can open a new ticket if need be, but that's also a regression in behavior/UX that we'd need to prioritize before v1.5 RC.

Reply:
Thinking: Do we need the same exception handling in all of our `@requires` decorators? (Is that ... a decorator for the decorators?)

Reply:
As long as the `postflight` decorator comes before our other custom decorators, it should catch anything raised.